### ISSUE TITLE
Corolis links don't quite work in Linux

### ISSUE THREAD

**Issue by Gaelmare** [jixxed/ed-odyssey-materials-helper#438]
I had to run 
```xdg-mime default edomh.desktop x-scheme-handler/edomh```
 in a terminal to correct the registered path of the helper. Looks like it provides the whole path when EDOMH registers the mime type, but at least Fedora 39 wants just the desktop application file name without a path.

Some tests:
working (after above command): 
```
$ xdg-mime query default x-scheme-handler/edomh
edomh.desktop
```
not working (after "Register application" in Settings>Handle edohm:// links): 
```
$ xdg-mime query default x-scheme-handler/edomh
/home/USERNAME/.local/share/applications/edomh.desktop 
```

Reproduction:
When I run this test command with an EDOMH URL from Corolis, I get the following error:
```
$ xdg-open edomh://coriolis/?eJ<BIGSTRING>
gio: edomh://coriolis/?eJ<BIGSTRING>: The specified location is not supported
```

This is not a new requirement per this 12 year old question: 
[xdg-open doesn't recognize custom protocol](https://www.linuxquestions.org/questions/linux-desktop-74/xdg-open-doesn%27t-recognize-custom-protocol-4175433062/)

**Comment by dmitriigo** [comment]
Hi. I had same issue. Please try to rename
$path_to_portable/bin/Elite Dangerous Odyssey Materials Helper
to
$path_to_portable/bin/EliteDangerousOdysseyMaterialsHelper
and
$path_to_portable/lib/app/Elite Dangerous Odyssey Materials Helper.cfg
to
$path_to_portable/lib/app/EliteDangerousOdysseyMaterialsHelper.cfg
and check again

**Comment by Gaelmare** [comment]
I did that, didn't help. Running the command listed fixed the problem.

**Comment by InnerBushman** [comment]
I have the same problem with path being added to desktop file name when registered. Re-registering it manually without the path works to point it to a correct target.
But I also have another problem with the generated .desktop file. The original file's exec field does not work at least on Manjaro. If i try
```xdg-open edomh://test```
I get an error
```Desktop file didn’t specify Exec field```
The original exec line is:
```Exec=/home/username/PathToEDOMH/bin/Elite\ Dangerous\ Odyssey\ Materials\ Helper %u```
And the modified by me, that work on Manjaro is:
```Exec='/home/username/PathToEDOMH/bin/Elite Dangerous Odyssey Materials Helper' %u```

Yet another thing is the file is generated without execute flag.

After doing some research and trying to cover as many bases as I could I landed ona file looking something like this:
```#!/usr/bin/env xdg-open
[Desktop Entry]
Name=Elite Dangerous Odyssey Materials Helper
Exec='/home/username/PathToEDOMH/bin/Elite Dangerous Odyssey Materials Helper' %u
Comment=Elite Dangerous Odyssey Material Helper
Type=Application
Terminal=false
Icon=package_development # just the icon I chose. would be nice to have icon file we could point to
MimeType=x-scheme-handler/edomh;
Hidden=false
NoDisplay=false
Categories=Game;
```
Some of the above is optional, but would be nice to have by default.

To sum it up:
- can we have registering without the path to desktop file?
- can we have Exec field that is unambiguous for most distros?
- can we have exec flag set when the file is created?

Cheers and Happy new year to everyone.

**Comment by dmitriigo** [comment]
This is the problem I wrote about earlier. Another option is to rename the files. You can also use a ready-made package from [aur](https://aur.archlinux.org/packages/ed-odyssey-materials-helper-bin) that contains a fix for this problem. 
I suggest the developer rename the files. This is the easiest way to solve the problem.

**Comment by Gaelmare** [comment]
So Fedora works with just the registration fix, no renames or changes to
desktop files. I'd like to see working installs out of the box for
everyone, whatever it takes. Spaces in filenames are sus anyway. ;)

**Comment by jixxed** [comment]
the spec also says double quotes :)

the issue is that every distro does it a little different. I can test on ubuntu(which is the only distro I officially support), but other distros are depending on the users providing clear information on commands that need to be executed.

you could try with `\s` instead of the `\ `. that could work a bit more distro independent: https://specifications.freedesktop.org/desktop-entry-spec/1.1/value-types.html

https://specifications.freedesktop.org/desktop-entry-spec/1.1/exec-variables.html


so far I've seen requests for fedora and arch, so I am willing to add in unofficial specific support for those in the code.

**Comment by InnerBushman** [comment]
> you could try with `\s` instead of the `\ `. that could work a bit more distro independent: 

The most distro independent is no space in file names in the first place. I hope you'll also agree and consider changing the file name. Pretty much everyone is shortening it to EDOMH so it's a very fitting name for the binary: "edomh.exe" short and to the point. The end user is almost never gonna be interacting with the binary anyway, that's what the desktop/shortcut files are for! :)

**Comment by jixxed** [comment]
changing the filename is a no-go, since it will break existing installations

**Comment by InnerBushman** [comment]
Can you explain how would that break existing installations if the auto-updater should handle file replacements and updating the shortcuts? portable instances are replaced as a whole anyway.
Can you elaborate what could go wrong and if there is really no way to work around it?

**Comment by jixxed** [comment]
the auto updater does not handle the desktop file. that file is only created by the register button in the app and is only intended to create a mapping for the edomh url. not to add it to a menu.
the auto updater does add itself to the menu and is the intended way to launch the application

**Comment by jixxed** [comment]
https://github.com/jixxed/ed-odyssey-materials-helper/releases/tag/2.131 contains an rpm build and a zip, in case rpm and deb don't do it.

there is now a registrationhandler that is OS specific. https://github.com/jixxed/ed-odyssey-materials-helper/blob/master/application/src/main/java/nl/jixxed/eliteodysseymaterials/service/registry/UbuntuRegistrationHandler.java

this is the default registration handler, but if anyone wants to add one for any other distro, go ahead with a PR and I'll include it.

**Comment by jixxed** [comment]
xdg-mime default edomh.desktop x-scheme-handler/edomh is now used, also works for ubuntu
